### PR TITLE
Fix Personalized search criterias

### PR DESCRIPTION
**Pull request by BB2A-Anthony**
# Fix #[*Personalized search criterias*]
Personalized search criterias is not add a search she replace search

### CODE REVIEW

File: htdocs/societe/list.php [hunk 1]
```diff
@@ -153,7 +153,7 @@
 if (($tmp = $langs->transnoentities("ProfId6".$mysoc->country_code)) && $tmp != "ProfId6".$mysoc->country_code && $tmp != '-') $fieldstosearchall['s.idprof6'] = 'ProfId6';
 if (!empty($conf->barcode->enabled)) $fieldstosearchall['s.barcode'] = 'Gencod';
 // Personalized search criterias. Example: $conf->global->THIRDPARTY_QUICKSEARCH_ON_FIELDS = 's.nom=ThirdPartyName;s.name_alias=AliasNameShort;s.code_client=CustomerCode'
-if (!empty($conf->global->THIRDPARTY_QUICKSEARCH_ON_FIELDS)) $fieldstosearchall = dolExplodeIntoArray($conf->global->THIRDPARTY_QUICKSEARCH_ON_FIELDS);
+if (!empty($conf->global->THIRDPARTY_QUICKSEARCH_ON_FIELDS)) $fieldstosearchall += dolExplodeIntoArray($conf->global->THIRDPARTY_QUICKSEARCH_ON_FIELDS);
```
Comment (eldy) [inline htdocs/societe/list.php:156]: Goal of option THIRDPARTY_QUICKSEARCH_ON_FIELDS is to be able to decide which param is used for quick search. Making this change allows only to complete, that is not enough.

Comment (bb2a) [reply htdocs/societe/list.php:156]: @eldy 
its not the comportement in V11

Comment (eldy) [reply htdocs/societe/list.php:156]: in v11, the code is 
if (!empty($conf->global->THIRDPARTY_QUICKSEARCH_ON_FIELDS)) $fieldstosearchall = dolExplodeIntoArray($conf->global->THIRDPARTY_QUICKSEARCH_ON_FIELDS);

So it means the fieldstosearch array is completely replaced with the expected values in the setup. So in v11 we are able to decide which field is used or not. This is the expected behavior.

Comment (bb2a) [reply htdocs/societe/list.php:156]: Ok so PR close

**Pull request closed without merging**